packages/gitbook/src/components/SiteSections/encodeClientSiteSections.ts

```diff
@@ -57,6 +61,33 @@ function encodeSection(context: GitBookSiteContext, section: SiteSection) {
     };
 }
 
+/**
+ * Test if a section should be included in the list of sections.
+ */
+function shouldIncludeSection(context: GitBookSiteContext, section: SiteSection) {
+    if (context.site.id !== 'site_JOVzv') {
+        return true;
+    }
+
+    // Testing for a new mode of navigation where the multi-variants section are hidden
+    // if they do not include an equivalent of the current site space.
+
+    // TODO: replace with a proper flag on the section
+    const withNavigateOnlyIfEquivalent = section.id === 'sitesc_4jvEm';
+
+    if (!withNavigateOnlyIfEquivalent) {
+        return true;
+    }
+
+    const { siteSpace: currentSiteSpace } = context;
+    if (section.siteSpaces.length === 1) {
+        return true;
+    }
```

Comment:
Not sure about that @SamyPesse

Reply:
It doesn't matter right now since this demo has multiple languages in all site sections; we can revisit it when we make it a proper option.

My thinking was: if there is only one site space, then it means it's not "multi-variants", so no point on filtering

Reply:
It also means, you select spanish on one section and all your site is only translated in english (without variants added yet), so you should hide english to respect the setting. But yeah we can revisit it.